_posts/2025-09-09-running-open-vsx-on-premises.adoc

```diff
@@ -0,0 +1,177 @@
+---
+title: "Running Open VSX On-Premises"
+layout: post
+author: Valerii Svydenko
+description: >-
+  Take Control of Your Extensions: Running Open VSX on-Premises with Eclipse Che
+categories: []
+keywords: ['Eclipse Open VSX', 'Eclipse Che', 'Extension Marketplace']
+slug: /@svor/running-open-vsx-on-premises
+---
+
+== Introduction
+
+Developers love extensions — they customize editors, add new languages, and streamline workflows.
+But in enterprise environments, relying on the public link:https://open-vsx.org[**Open VSX registry**] can introduce challenges:
+air-gapped environments, compliance rules, or simply the need for a curated list of approved tools.
+
+That’s where running **Open VSX On-Premises** comes in.
+In this post, we’ll explore how {prod} makes it easy to deploy and configure your own extension registry, using nothing more than an {prod} workspace.
+
+WARNING: ⚠️ This flow is supported only on **OpenShift clusters where {prod} is deployed**. It will not work on plain Kubernetes.
```

Comment:
That's a pretty strong statement, do not we / openvsx have any instructions for vanilla k8s?

Reply:
@ibuziuk not yet https://github.com/eclipse-che/che/issues/23260